service/controller/v3/resource/vpngateway/helper.go

```diff
@@ -0,0 +1,33 @@
+package vpngateway
+
+import (
+	"math/rand"
+
+	"github.com/giantswarm/microerror"
+)
+
+// toVPNGatewayConnections convert v to network.VirtualNetworkGatewayConnection.
+// If v is nil and empty network.VirtualNetworkGatewayConnection is returned.
+func toVPNGatewayConnections(v interface{}) (connections, error) {
+	if v == nil {
+		return connections{}, nil
+	}
+
+	vpnGatewayConnections, ok := v.(connections)
+	if !ok {
+		return connections{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", connections{}, v)
+	}
+
+	return vpnGatewayConnections, nil
+}
+
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"
+
+func randStringBytes(n int) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Intn(len(letterBytes))]
```

Comment:
Are you sure it's initialized? https://golang.org/pkg/math/rand/#Seed

Please grep the code and check. If not lets do it in some meaningful file like main.go

Reply:
done

Reply:
https://github.com/giantswarm/azure-operator/pull/278/commits/a4f92645e87f4c41b59536fe66a3a9cffba62b26